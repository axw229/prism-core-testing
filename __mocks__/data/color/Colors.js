// @flow
import cloneDeep from 'lodash/cloneDeep'
import sample from 'lodash/sample'

import type { ColorSetPayload, Color } from '../../../src/shared/types/Colors'

export const Colors: ColorSetPayload = {
  "Red": [
    [
      {
        "colorNumber": "6561",
        "coordinatingColors": {
          "coord2ColorId": 1926,
          "coord1ColorId": 2250,
          "whiteColorId": 2681
        },
        "description": [
          "Subdued",
          "Muted",
          "Brilliant",
          "Radiant"
        ],
        "id": "2248",
        "isExterior": true,
        "isInterior": true,
        "name": "Teaberry",
        "rgb": 15454683,
        "lrv": 68.551,
        "colorFamilyNames": [
          "Red"
        ],
        "brandKey": "SW",
        "red": 235,
        "green": 209,
        "blue": 219,
        "hue": 0.935897435897436,
        "saturation": 0.39393939393939387,
        "lightness": 0.8705882352941177,
        "hex": "#ebd1db",
        "isDark": false,
        "cssrgb": "rgb(235,209,219)",
        "storeStripLocator": "102-C1",
        "similarColors": [
          "2255",
          "2262",
          "2269",
          "2662",
          "2535",
          "2276",
          "2658",
          "1974",
          "2543",
          "2531"
        ]
      },
      {
        "colorNumber": "6568",
        "coordinatingColors": {
          "coord2ColorId": 11244,
          "coord1ColorId": 2687,
          "whiteColorId": 2681
        },
        "description": [
          "Subdued",
          "Muted",
          "Brilliant",
          "Radiant"
        ],
        "id": "2255",
        "isExterior": true,
        "isInterior": true,
        "name": "Lighthearted Pink",
        "rgb": 15586781,
        "lrv": 70.758,
        "brandedCollectionIds": [
          31797,
          31791
        ],
        "brandedCollectionNames": [
          "Pottery Barn Fall/Winter 2018",
          "Pottery Barn Fall/Winter 2018"
        ],
        "colorFamilyNames": [
          "Red"
        ],
        "brandKey": "SW",
        "red": 237,
        "green": 213,
        "blue": 221,
        "hue": 0.9444444444444445,
        "saturation": 0.3999999999999999,
        "lightness": 0.8823529411764706,
        "hex": "#edd5dd",
        "isDark": false,
        "cssrgb": "rgb(237,213,221)",
        "storeStripLocator": "103-C1",
        "similarColors": [
          "2248",
          "2262",
          "2269",
          "2276",
          "1974",
          "1981",
          "2658",
          "1995",
          "2662",
          "1988"
        ]
      }
    ]
  ]
}

export function getColors (): ColorSetPayload {
  return cloneDeep(Colors)
}

export function getColor (): Color | void {
  const colors = getColors()
  const fam1 = Object.keys(colors)[0]
  if (colors[fam1]) {
    return sample(colors[fam1][0])
  }
}
