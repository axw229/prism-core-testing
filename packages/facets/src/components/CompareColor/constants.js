export const baseClass = 'color__comparison'
export const containerClass = `${baseClass}__container`
export const containerHeaderClass = `${baseClass}__container__header`
export const containerHeaderButtonClass = `${baseClass}__container__header__button`
export const wrapperClass = `${baseClass}__wrapper`
export const prevBtnWrapperClass = `${baseClass}__prev-btn__wrapper`
export const buttonsClass = `${baseClass}__buttons`
export const buttonsVisibleClass = `${baseClass}__buttons--visible`
export const nextBtnWrapperClass = `${baseClass}__next-btn__wrapper`
export const backgroundColorClass = `${baseClass}__background__color`
export const queueWrapperClass = `${baseClass}__queue__wrapper`
