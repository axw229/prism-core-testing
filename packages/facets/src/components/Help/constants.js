export const baseClass = `cvw-help`
export const wrapper = `${baseClass}__wrapper`
export const tabsContainer = `${baseClass}__tabs-container`
export const contentWrapper = `${baseClass}__content-wrapper`
export const helpContent = `${baseClass}__help-content`
export const helpContentHide = `${helpContent}--hide`
export const contentHeader = `${baseClass}__content-header`
export const contentDetails = `${baseClass}__content-details`
export const subContent = `${baseClass}__subcontent`
export const subContentHeader = `${baseClass}__subContent-header`
export const subContentDetails = `${baseClass}__subContent-details`
export const helpIcons = `${baseClass}__help-icons`
export const helpImages = `${baseClass}__help-images`
export const helpImagesMobile = `${baseClass}__help-images-mobile`
export const iconWrap = `${baseClass}__icon-wrap`
export const iconWrapUndoRedo = `${baseClass}__icon-wrap-undoredo`
export const iconInfo = `${baseClass}__icon-info`
export const secondIcon = `${baseClass}__second-icon`
export const SCROLL_SPEED = 500
