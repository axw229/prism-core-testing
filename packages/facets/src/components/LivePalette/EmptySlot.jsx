import React from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import 'src/providers/fontawesome/fontawesome'

function EmptySlot () {
  return (
    <div className='prism-live-palette__slot prism-live-palette__slot--empty'>
      <div className='prism-live-palette__slot__guts'>
        <FontAwesomeIcon className='prism-live-palette__slot__icon' icon={['fal', 'plus-circle']} size='2x' />
      </div>
    </div>
  )
}

export default EmptySlot
