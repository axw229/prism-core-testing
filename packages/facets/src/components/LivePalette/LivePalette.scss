.prism-live-palette {
  display: flex;
  background-color: var(getThemeColor('white'));
  padding: $livePalette__chip-side-margin;
  font-family: var(getVarName('typography.titleFontFamily'));
  @include respond-above(sm) {
    border: 1px solid $color-grey;
  }

  &-simple {
    border: none;
  }

  &__header {
    display: flex;
    flex-direction: column;
    justify-content: flex-end;
    display: none;
    @include respond-above(sm) {
      display: flex;
    }
    &__name {
      display: none;
      text-align: center;
      @include respond-above(sm) {
        display: block;
        margin: auto 0.5em;
      }
    }
    &__compare-button{
      &--hide{
        display: none;
      }
      padding: 0.5em 1em;
      margin-right: 5px;
      font-family: var(getVarName('typography.titleFontFamily'));
      font-size: .8em;
      outline:none;
      text-transform: var(getVarName('typography.buttonTextTransform'));
      font-weight: var(getVarName('typography.buttonFontWeight'));
      border: 1px solid var(getThemeColor('buttonBorder'));
      background-color: var(getThemeColor('buttonBgColor'));
      color: var(getThemeColor('buttonColor'));

      &:hover{
        border: 1px solid var(getThemeColor('buttonHoverBorder'));
        background-color: var(getThemeColor('buttonHoverBgColor'));
        color: var(getThemeColor('buttonHoverColor'));
        cursor: pointer;
      }
    }
  }

  &__height {
    @include respond-above(sm) {
      height: 4rem !important;
    }
    @include respond-above(md) {
      height: 4.625rem !important;
    }
  }

  &__slot {
    $lp-icon-size: 24px;

    width: inherit;
    height: 100%;

    &:focus {
      outline-style: solid;
      outline-color: var(getThemeColor('secondary'));
      outline-width: 2px;
    }

    &--empty {
      display: flex;
      justify-content: center;
      align-items: center;
      width: inherit;
      height: 100%;
      cursor: default;
      background-color: var(getThemeColor('light'));
    }

    &__label {
      display: none;
      font-size: #{(9/16)}rem;
      text-align: center;

      @include respond-above(md) {
        display: block;
      }

      &--text {
        width: 9rem;
        margin: 0.2rem;
        font-family: var(getVarName('typography.bodyFontFamily'));
        font-style: var(getVarName('typography.titleFontStyle'));
        font-size: .875rem;
        line-height: 1.2rem;
      }
    }

    &__icon {
      width: $lp-icon-size !important;
      height: $lp-icon-size !important;
      margin: 5px;
      color: var(getThemeColor('light.contrast'));

      &--empty { opacity: .3 }
    }

    &__button {
      position: absolute;
      top:50%;
      left:50%;
      transform: translate(-50%, -50%);
      outline:none;
    }
  }

  &__info {
    margin: 0 .125rem !important;
    font-size: 1.4rem !important;

    &--outlined {
      display: flex;
      justify-content: center;
      align-items: center;
      width: 1rem;
      height: 1rem;
      outline: 2px solid;

      &-icon { font-size: .8rem !important }
    }

    &:focus {
      outline-style: solid;
      outline-color: var(getThemeColor('secondary'));
      outline-width: 2px;
    }
  }

  &__trash {
    @include respond-above(sm) {
      margin-left: .25rem !important;
    }

    &:focus {
      outline-style: solid;
      outline-color: var(getThemeColor('secondary'));
      outline-width: 2px;
    }

    &--icon { font-size: 20px }
  }

  &__name-number {
    display: flex;
    flex-direction: column-reverse;
  }

  &__color-number {
    display: block;
    font-size: .8rem;
    font-weight: normal;
    color: inherit;
    white-space: nowrap;
  }

  &__color-name {
    display: block;
    font-size: .8rem;
    color: inherit;
    @include respond-above(sm) {
      font-weight: 600;
    }
  }

  &__color-description {
    clip: rect(1px, 1px, 1px, 1px); // For screen readers only
    font-size: 0.7em;
    display: none;
    position: absolute;
    bottom: 0.5em;
    color: inherit;
    @include respond-above(sm) {
      display: block;
    }
  }

  &-house-shaped {
    &__name-number {
      display: flex;
      flex-direction: column-reverse;
      line-height: 1.2rem;
    }

    &__color-name {
      font-family: var(getVarName('typography.bodyFontFamily'));
      font-style: italic;
      white-space: nowrap;
    }

    &__color-number {
      font-size: .875rem;
      line-height: .9rem;
      white-space: nowrap;
    }

    &__height {
      @include respond-below(sm) {
        height: 5.6rem !important;
      }
    }
  }
}
