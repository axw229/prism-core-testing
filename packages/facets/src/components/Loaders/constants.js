// @flow
// the 5 colors used by loaders in their default state; taken from SW CVW circa 2019
export const DEFAULT_COLORS = [
  '#00b9e4',
  '#69be28',
  '#ffb612',
  '#e17000',
  '#b71234'
]
// how many colors there are (this is sort of magic, so don't touch it)
export const NUM_COLORS = DEFAULT_COLORS.length
// how many generated colors are allowed to be dark
export const NUM_DARKS = Math.floor(NUM_COLORS / 2)
// how many generated colors should be light
export const NUM_LIGHTS = NUM_COLORS - NUM_DARKS
// default brightness step between generated colors
export const GENERATED_COLOR_BRIGHTNESS_STEP = 10
// % of brightness spectrum edges considered to be too dark or too light
export const GENERATED_COLOR_EDGE_PCT = 0.1
