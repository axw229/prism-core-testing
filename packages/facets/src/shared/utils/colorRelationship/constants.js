export const FULL = 360
export const HALF = FULL / 2
export const QUARTER = HALF / 2
export const NUM_CLUSTERS = 12
export const CLUSTER_SIZE = FULL / NUM_CLUSTERS
