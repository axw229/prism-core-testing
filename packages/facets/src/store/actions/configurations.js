// @flow
import axios from 'axios'

import { CONFIG_ENDPOINT } from 'constants/endpoints'
import { generateBrandedEndpoint } from 'src/shared/helpers/DataUtils'

export const REQUEST_CONFIGURATION: string = 'REQUEST_CONFIGURATION'
const requestConfiguration = () => {
  return {
    type: REQUEST_CONFIGURATION,
    payload: {
      loadingConfiguration: true
    }
  }
}

export const RECEIVE_CONFIGURATION: string = 'RECEIVE_CONFIGURATION'
const receiveConfiguration = (configurationData: Number) => {
  return {
    type: RECEIVE_CONFIGURATION,
    payload: {
      loadingConfiguration: false,
      configuration: configurationData
    }
  }
}

export const LOAD_ERROR: string = 'LOAD_ERROR'
const loadError = (err) => {
  return {
    type: LOAD_ERROR,
    payload: err
  }
}

export const loadConfiguration = (brandId: string) => {
  return (dispatch: Function, getState: Function) => {
    dispatch(requestConfiguration())
    const { current } = getState().language

    axios
      .get(generateBrandedEndpoint(CONFIG_ENDPOINT, brandId.toLowerCase(), { language: current })) // TODO: remove this toLowecase()
      .then(r => r.data)
      .then(config => {
        dispatch(receiveConfiguration(config))
      })
      .catch(err => {
        dispatch(loadError(err))
        console.error(`There was an error loading the configuration for ${brandId}.`, err)
      })
  }
}
