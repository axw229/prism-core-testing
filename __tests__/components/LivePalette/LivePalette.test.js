import React from 'react'
import { shallow } from 'enzyme'
import { LivePalette } from 'src/components/LivePalette/LivePalette'
import * as Colors from '__mocks__/data/color/Colors'
import cloneDeep from 'lodash/cloneDeep'
import EmptySlot from 'src/components/LivePalette/EmptySlot'
import { LP_MAX_COLORS_ALLOWED } from 'constants/configurations'
import { Link } from 'react-router-dom'

const activeSlotHOC = 'Connect(DragSource(DropTarget(ActiveSlot)))'
const colors = Colors.getlpColors()
const defaultProps = cloneDeep(colors)

const createLivePalette = (props = {}) => {
  return shallow(<LivePalette {...defaultProps} {...props} />)
}

// snapshot testing
describe('snapshot testing for LivePalette component ', () => {
  it('LivePalette component should match snapshot', () => {
    const wrapper = createLivePalette()
    expect(wrapper).toMatchSnapshot()
  })
})

// rendering test
describe('rendering testing for LivePalette component ', () => {
  it('LivePalette component rendering correctly', () => {
    const wrapper = createLivePalette()
    expect(wrapper.find('.prism-live-palette').length).toBeGreaterThan(0)
  })

  it('LivePalette should rendering numbers of active slots correctly', () => {
    const wrapper = createLivePalette()
    const activeSlotCount = wrapper.find(activeSlotHOC).length

    if (activeSlotCount > 0) {
      expect(activeSlotCount).toBeGreaterThan(0)
      if (activeSlotCount > LP_MAX_COLORS_ALLOWED) {
        expect(activeSlotCount).toEqual(LP_MAX_COLORS_ALLOWED)
      }
    } else {
      expect(activeSlotCount).toEqual(0)
    }
  })

  it('LivePalette should rendering numbers of Empty slots correctly', () => {
    const wrapper = createLivePalette()
    const isHaveEmptySlot = (LP_MAX_COLORS_ALLOWED - 1) - wrapper.find(activeSlotHOC).length
    const emptySlotCount = wrapper.find(EmptySlot).length

    if (isHaveEmptySlot > 0) {
      expect(emptySlotCount).toBeGreaterThan(0)
    } else {
      expect(emptySlotCount).toEqual(0)
    }
  })

  it('LivePalette should rendering numbers of Link correctly', () => {
    const wrapper = createLivePalette()
    const countLinks = wrapper.find(Link).length
    if (defaultProps.colors.length < LP_MAX_COLORS_ALLOWED) {
      expect(countLinks).toBeGreaterThan(0)
    } else {
      expect(countLinks).toEqual(0)
    }
  })
})

// event test
describe('event testing for LivePalette componenet', () => {
  it('onClick function of activeSlotHOC should be called with parameter colors', () => {
    const spy = jest.fn()
    const wrapper = createLivePalette({ activateColor: spy })
    wrapper.find(activeSlotHOC).forEach((el, index) => {
      el.simulate('click')
      expect(spy).toBeCalled()
    })
  })
})
