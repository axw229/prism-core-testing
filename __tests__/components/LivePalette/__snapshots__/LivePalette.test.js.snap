// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing for LivePalette component  LivePalette component should match snapshot 1`] = `
<div
  className="prism-live-palette"
>
  <LivePalletteModal
    isActive={false}
  />
  <div
    className="prism-live-palette__header"
  >
    <span
      className="prism-live-palette__header__name"
    >
      <FormattedMessage
        id="PALETTE_TITLE"
        values={Object {}}
      />
    </span>
    <button
      className="prism-live-palette__header__compare-button"
      onClick={[Function]}
    >
      Compare Color
    </button>
  </div>
  <div
    className="prism-live-palette__list"
  >
    <Connect(DragSource(DropTarget(ActiveSlot)))
      active={false}
      color={
        Object {
          "blue": 108,
          "brandKey": "SW",
          "brandedCollectionNames": Array [
            "Interior Historic Collection",
          ],
          "colorFamilyNames": Array [
            "Historic Color",
          ],
          "colorNumber": "0001",
          "coordinatingColors": Object {
            "coord1ColorId": "11290",
            "coord2ColorId": "11296",
          },
          "description": Array [
            "Dull",
            "Fairly Bright",
          ],
          "green": 118,
          "hex": "#94766c",
          "hue": 0.041666666666666685,
          "id": "1544",
          "isDark": false,
          "isExterior": true,
          "isInterior": true,
          "lightness": 0.5019607843137255,
          "lrv": 20.427,
          "name": "Mulberry Silk",
          "red": 148,
          "saturation": 0.15748031496062997,
          "similarColors": Array [
            "1733",
            "1719",
            "1747",
            "1754",
            "1712",
            "1775",
            "1740",
            "2831",
            "1768",
            "11269",
          ],
        }
      }
      index={0}
      isCompareColor={false}
      key="1544"
      moveColor={[Function]}
      node={
        Object {
          "current": null,
        }
      }
      onClick={[Function]}
    />
    <Connect(DragSource(DropTarget(ActiveSlot)))
      active={false}
      color={
        Object {
          "blue": 174,
          "brandKey": "SW",
          "brandedCollectionNames": Array [
            "Lifestyle",
            "HGTV Homeâ„¢ by Sherwin-Williams",
          ],
          "colorFamilyNames": Array [
            "Blue",
          ],
          "colorNumber": "6759",
          "coordinatingColors": Object {
            "coord1ColorId": "2689",
            "coord2ColorId": "11355",
            "whiteColorId": "2436",
          },
          "description": Array [
            "Subdued",
            "Muted",
            "Fairly Bright",
          ],
          "green": 185,
          "hex": "#75b9ae",
          "hue": 0.47303921568627455,
          "id": "2446",
          "isDark": false,
          "isExterior": true,
          "isInterior": true,
          "lightness": 0.592156862745098,
          "lrv": 41.426,
          "name": "Cooled Blue",
          "red": 117,
          "saturation": 0.3269230769230769,
          "similarColors": Array [
            "2439",
            "11224",
            "11223",
            "1618",
            "2447",
            "2453",
            "2173",
            "2624",
            "11232",
            "2151",
          ],
          "storeStripLocator": "163-C4",
        }
      }
      index={1}
      isCompareColor={false}
      key="2446"
      moveColor={[Function]}
      node={
        Object {
          "current": null,
        }
      }
      onClick={[Function]}
    />
    <Link
      className="prism-live-palette__slot prism-live-palette__slot--add"
      to="/active/color-wall"
    >
      <FontAwesomeIcon
        border={false}
        className="prism-live-palette__icon"
        color="#0069af"
        fixedWidth={false}
        flip={null}
        icon={
          Array [
            "fal",
            "plus-circle",
          ]
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size="2x"
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
      <FormattedMessage
        id="ADD_A_COLOR"
        values={Object {}}
      >
        <Component />
      </FormattedMessage>
    </Link>
    <EmptySlot
      key="0"
    />
    <EmptySlot
      key="1"
    />
    <EmptySlot
      key="2"
    />
    <EmptySlot
      key="3"
    />
    <EmptySlot
      key="4"
    />
  </div>
  <aside
    aria-live="assertive"
    className="prism-live-palette__color-description"
  />
</div>
`;
